From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Mon, 13 Jul 2026 16:51:53 +0800
Subject: [PATCH 15/26] native: Allow interface name mismatch for Waydroid
 framework

Revert this when tools/interfaces/*.py migrates to the new framework name

Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 libs/binder/Parcel.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libs/binder/Parcel.cpp b/libs/binder/Parcel.cpp
index b6284f28c..4882f9528 100644
--- a/libs/binder/Parcel.cpp
+++ b/libs/binder/Parcel.cpp
@@ -1241,6 +1241,10 @@ bool Parcel::enforceInterface(const char16_t* interface,
             // this point, but it would be extremely fragile. It's more important that
             // we fuzz with the above things read from the Parcel.
             return true;
+        } else if (strncmp(String8(interface, len), "id.waydro.", 10) == 0 &&
+                strncmp(String8(parcel_interface, parcel_interface_len), "lineageos.waydroid.", 19) == 0) {
+            // Allow interface name mismatch for Waydroid framework
+            return true;
         } else {
             ALOGW("**** enforceInterface() expected '%s' but read '%s'",
                   String8(interface, len).c_str(),
